django/db/models/sql/compiler.py

```diff
@@ -526,10 +527,13 @@ def as_sql(self, with_limits=True, with_col_aliases=False):
                         raise NotSupportedError('SKIP LOCKED is not supported on this database backend.')
                     elif of and not self.connection.features.has_select_for_update_of:
                         raise NotSupportedError('FOR UPDATE OF is not supported on this database backend.')
+                    elif no_key and not self.connection.features.has_select_for_no_key_update:
```

Comment:
I think you miss a test for this change.

Reply:
Sorry, I didn't see this message. I just rebased and force-pushed a version which includes a test for this change.